service/chartconfig/v1/resource/chart/desired.go

```diff
@@ -2,8 +2,23 @@ package chart
 
 import (
 	"context"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/chart-operator/service/chartconfig/v1/key"
 )
 
 func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	chartState := ChartState{
```

Comment:
It would be good to propagate pointers. Then we can use nil as zero value. At the moment this is inconsistent and the error returns should actually return the empty struct.

Reply:
Good point. Done.